src/Tokenizer/Tokens.php

```diff
@@ -1088,16 +1088,18 @@ public function clearRange(int $indexStart, int $indexEnd): void
      */
     public function isMonolithicPhp(): bool
     {
-        $size = $this->count();
-
-        if (0 === $size) {
+        if (0 === $this->count()) {
             return false;
         }
 
-        if ($this->isTokenKindFound(T_INLINE_HTML)) {
+        if ($this->countTokenKind(T_INLINE_HTML) > 1) {
             return false;
         }
 
+        if (1 === $this->countTokenKind(T_INLINE_HTML)) {
+            return 1 === Preg::match('/^#!.*$/m', $this[0]->getContent());
```

Comment:
not 100% sure about the `m` modifier, I was thing about something like:

```
yield [false, "#!/usr/bin/php\nFoo bar<?php\n"];
yield [false, "#!/usr/bin/env php -n \nFoo bar\n<?php\n"];
```

Reply:
:+1:
plus the variants with concrete php, eg php7.4, php7.4-cli, php8 etc

Reply:
Definitely, I don't know what I was thinking, I've named test case "this-is too-much" and made is passing check 🤦🏼